module/scheduler_components/simple_remaster_manager.cpp

```diff
@@ -63,36 +63,26 @@ SimpleRemasterManager::RemasterOccured(Key remaster_key, uint32_t /* remaster_co
   return result;
 }
 
-RemasterOccurredResult SimpleRemasterManager::ReleaseTransaction(TxnId txn_id) {
-  // Collect all local logs
-  unordered_set<uint32_t> replicas;
-  for (auto& queue_pair : blocked_queue_) {
-    if (!queue_pair.second.empty()) {
-      replicas.insert(queue_pair.first);
-    }
-  }
-  return ReleaseTransaction(txn_id, replicas);
-}
-
 RemasterOccurredResult
-SimpleRemasterManager::ReleaseTransaction(TxnId txn_id, const unordered_set<uint32_t>& replicas) {
-  for (auto& p : replicas) {
+SimpleRemasterManager::ReleaseTransaction(const TransactionHolder* txn_holder) {
+  auto txn_id = txn_holder->GetTransaction()->internal().id();
+  for (auto& p : txn_holder->InvolvedReplicas()) {
```

Comment:
nit: replicas are integers so not necessary to use `auto&`
Also why name it `p` if you're referring to a replica?